### PR TITLE
CwdAwareHinter: remove cwd filter when there are no results

### PR DESCRIPTION
**Pull request by p00f**
closes https://github.com/nushell/nushell/issues/10921

### CODE REVIEW

File: src/hinter/cwd_aware.rs [hunk 1]
```diff
@@ -39,15 +39,29 @@ impl Hinter for CwdAwareHinter {
...
-                .expect("todo: error handling")
-                .get(0)
-                .map_or_else(String::new, |entry| {
-                    entry
-                        .command_line
-                        .get(line.len()..)
-                        .unwrap_or_default()
-                        .to_string()
-                })
+                .unwrap_or_else(|_| vec![]);
```
Comment (sholderbach) [inline src/hinter/cwd_aware.rs:42]: Good call! That should eliminate a possible failure path! And yeah having no hint wouldn't be a problem if there is an error internally.

As `Vec` should implement `Default` you could also write

```suggestion
                .unwrap_or_default();
```

**Review by sholderbach — APPROVED** [review]
Nice!

File: src/hinter/cwd_aware.rs (hunk 1 above)
Comment (p00f) [reply src/hinter/cwd_aware.rs]: Thanks, done

**Comment by p00f** [comment]
I just saw that the next release is tomorrow so I wanted to get this in

**Comment by sholderbach** [comment]
Tested it with the `cwd_aware_hinter` demo and works like a charm. 

As we made it compulsory for the sqlite history let's get this qol improvement in.

**Pull request merged**